openapi-rs/api/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+pub struct InferRequestOptions {
+    pub max_tokens: Option<u32>,
+    pub repeat_penalty: Option<f32>,
+    pub repeat_penalty_last_n_token_count: Option<u32>,
+    pub temperature: Option<f32>,
+    pub top_k: Option<u32>,
+    pub top_p: Option<f32>,
+}
+
+impl Default for InferRequestOptions {
+    fn default() -> Self {
+        InferRequestOptions {
+            max_tokens: Some(DEFAULT_MAX_TOKENS),
+            repeat_penalty: Some(DEFAULT_REPEAT_PENALTY),
+            repeat_penalty_last_n_token_count: Some(DEFAULT_REPEAT_PENALTY_LAST_N_TOKEN_COUNT),
+            temperature: Some(DEFAULT_TEMPERATURE),
+            top_k: Some(DEFAULT_TOP_K),
+            top_p: Some(DEFAULT_TOP_P),
+        }
+    }
+}
+
+impl Into<spin_sdk::llm::InferencingParams> for InferRequestOptions {
+    fn into(self) -> spin_sdk::llm::InferencingParams {
+        spin_sdk::llm::InferencingParams {
+            //TODO: set constants for these defaults
+            max_tokens: self.max_tokens.unwrap_or(DEFAULT_MAX_TOKENS),
+            repeat_penalty: self.repeat_penalty.unwrap_or(DEFAULT_REPEAT_PENALTY),
+            repeat_penalty_last_n_token_count: self
+                .repeat_penalty_last_n_token_count
+                .unwrap_or(DEFAULT_REPEAT_PENALTY_LAST_N_TOKEN_COUNT),
+            temperature: self.temperature.unwrap_or(DEFAULT_TEMPERATURE),
+            top_k: self.top_k.unwrap_or(DEFAULT_TOP_K),
+            top_p: self.top_p.unwrap_or(DEFAULT_TOP_P),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InferResponse {
+    pub text: String,
+    pub usage: InferResponseUsage,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InferResponseUsage {
+    pub prompt_token_count: u32,
+    pub generated_token_count: u32,
+}
+
+impl From<spin_sdk::llm::InferencingResult> for InferResponse {
+    fn from(result: spin_sdk::llm::InferencingResult) -> Self {
+        InferResponse {
+            text: result.text,
+            usage: InferResponseUsage {
+                prompt_token_count: result.usage.prompt_token_count,
+                generated_token_count: result.usage.generated_token_count,
+            },
+        }
+    }
+}
+
+#[http_component]
+fn handle_api(req: Request) -> Result<Response> {
+    let component_route = req
+        .headers()
+        .get("spin-component-route")
+        .unwrap()
+        .to_str()?;
+    let mut router = Router::new();
+    router.post(&format!("{}/infer", component_route), handle_infer);
+    router.handle(req)
+}
+
+fn handle_infer(req: Request, _params: Params) -> Result<Response> {
+    // parse the request
+    let request: InferRequest =
+        serde_json::from_slice(req.body().as_ref().unwrap_or(&Bytes::new()))?;
+    let model: spin_sdk::llm::InferencingModel = match request.model.as_deref() {
+        Some("llama2-chat") => spin_sdk::llm::InferencingModel::Llama2Chat,
+        Some("llama2-code") => spin_sdk::llm::InferencingModel::CodellamaInstruct,
```

Comment:
Why rename the model here?

Reply:
I think I just did it for consistency before I understood that the names corresponded to the model's file name. Happy to rename it to `codellama-instruct` or whatever makes sense.